File: lib/src/main/java/com/tersesystems/echopraxia/LoggerFactory.java
package com.tersesystems.echopraxia;

import com.tersesystems.echopraxia.core.CoreLogger;
import org.slf4j.ILoggerFactory;

public final class LoggerFactory {

    public static Logger getLogger(Class<?> clazz) {
        return getLogger(clazz.getName());
    }

    public static Logger getLogger(String name) {
        ILoggerFactory factory = org.slf4j.LoggerFactory.getILoggerFactory();
        org.slf4j.Logger logger = factory.getLogger(name);
        return new Logger(new CoreLogger.Impl(logger));
    }

    public Logger getLogger(LoggerResolver resolver) {
        return resolver.get(this);
    }
}

File: scala/api/src/main/scala/com/tersesystems/echopraxia/sapi/LoggerFactory.scala
package com.tersesystems.echopraxia.sapi

import com.tersesystems.echopraxia.core.{Caller, CoreLoggerFactory}
import com.tersesystems.echopraxia.sapi.support.DefaultLoggerMethods

/**
 * A logger factory that returns a logger that can use tuples and scala types.
 */
object LoggerFactory {
  val FQCN: String = classOf[DefaultLoggerMethods[_]].getName

  val fieldBuilder: FieldBuilder = new FieldBuilder {}

  def getLogger(name: String): Logger[FieldBuilder] = {
    val core = CoreLoggerFactory.getLogger(FQCN, name)
    new Logger(core, fieldBuilder)
  }

  def getLogger(clazz: Class[_]): Logger[FieldBuilder] = {
    val core = CoreLoggerFactory.getLogger(FQCN, clazz.getName)
    new Logger(core, fieldBuilder)
  }

  def getLogger: Logger[FieldBuilder] = {
    val core = CoreLoggerFactory.getLogger(FQCN, Caller.resolveClassName)
    new Logger(core, fieldBuilder)
  }
}
